solver/src/settlement_submission.rs

```diff
@@ -143,3 +172,19 @@ async fn tenderly_link(
         network_id
     ))
 }
+
+async fn recover_gas_price_from_pending_transaction(
+    web3: &Web3<DynTransport>,
+    address: &H160,
+) -> Result<Option<U256>> {
+    let block = web3
+        .eth()
+        .block_with_txs(BlockId::Number(BlockNumber::Pending))
+        .await?
+        .ok_or_else(|| anyhow!("empty block"))?;
+    let transaction = block
+        .transactions
+        .iter()
+        .find(|transaction| transaction.from == *address);
```

Comment:
Should we also check nonce here to make sure we are finding the right transaction?

Reply:
The nonce would only be wrong if there are multiple pending transactions right? I felt we could ignore that case for now because it should never happen. If we wanted to handle it we could make sure to always take over the lowest pending nonce.

Reply:
We need to outbid the gas price of the transaction with the same nonce so even if we had more than one pending tx I think the nonce == check would be correct (instead of checking the gas price of the lowest pending nonce):

> "Transaction gas price supplied is too low. There is another transaction with same nonce in the queue. Try increasing the gas price or incrementing the nonce.",

It's true that currently this is not necessary but it feels like a very small addition to pass the nonce and do the check. Given this code path is already an edge case it is risky to forget/not test this if we want to handle multiple parallel settlement transactions.

Reply:
added